mac/Keyman4MacIM/Keyman4MacIM/KMPackage.m

```diff
@@ -82,4 +74,51 @@ -(BOOL)readFromData:(NSData *)data ofType:(NSString *)typeName error:(NSError *
     [[NSFileManager defaultManager] removeItemAtPath:tempFile error:nil];
     return didUnzip;
 }
+
+-(BOOL)userConfirmsInstallationOfPackageFile:(NSString *)fileName {
+    NSAlert *alert = [[NSAlert alloc] init];
+    [alert addButtonWithTitle:NSLocalizedString(@"Yes", @"Alert button")];
+    [alert addButtonWithTitle:NSLocalizedString(@"No", @"Alert button")];
+    [alert setMessageText:NSLocalizedString(@"Install Keyman Package?", @"Alert message text when user double-clicks a KMP file.")];
+    NSString *info = NSLocalizedString(@"Do you want the Keyman Input Method to install this Package?\rFile: %@", @"Alert informative text when user double-clicks a KMP file. Parameter is the name of the KMP file.");
+    info = [NSString localizedStringWithFormat:info, fileName];
+    [alert setInformativeText:info];
+    [alert setAlertStyle:NSInformationalAlertStyle];
+    
+    if ([self.AppDelegate debugMode])
+        NSLog(@"Asking user to confirm installation...");
+    
+    BOOL result = [alert runModal] == NSAlertFirstButtonReturn;
+    return result;
+}
+
+// This tells the system that we aren't reading the Keyman Package file lazily.
+-(BOOL) isEntireFileLoaded {
+    return YES;
+}
+
+//-(BOOL)readFromData:(NSData *)data ofType:(NSString *)typeName error:(NSError * _Nullable __autoreleasing *)outError {
```

Comment:
It'd be good to remove this commented code before merge.

Reply:
Thanks for the reminder. I wanted to include it in an intermediate commit just in case I ever needed to go back.